File: TypeScript/Codewars/src/resolutions/Tribonacci Sequence.ts
export function tribonacci([a, b, c]: [number, number, number], n: number): number[] {
	const output = [a, b, c];

	output.push(a + b + c);

	for (let index = 3; index < n - 1; index++) {
		const value = output[index] + output[index - 1] + output[index - 2];
		output.push(value);
	}

	return output.slice(0, n);
}

console.log(tribonacci([1, 1, 1], 1));

File: Csharp/CodeWars/src/Completed/TribonacciSequence.cs
namespace CodeWars.Completed;

public static class TribonacciSequence
{
    public static double[] Tribonacci(double[] signature, int n)
    {
        List<double> output = [.. signature, signature.Sum()];

        for (int i = 3; i < n - 1; i++)
            output.Add(output[i] + output[i - 1] + output[i - 2]);

        return [.. output.Take(n)];
    }
}
